packages/@storefront/flux-capacitor/src/core/adapters/search.ts

```diff
@@ -86,21 +87,32 @@ namespace Adapter {
   };
 
   export const pruneRefinements = (navigations: Store.Navigation[], state: Store.State): Store.Navigation[] => {
-    const max = ConfigAdapter.extractMaxRefinements(Selectors.config(state));
-    return max ? navigations.map((navigation) => {
-      const show = navigation.selected.slice(0, max);
-      for (let i = 0; i < navigation.refinements.length && show.length < max; i++) {
-        if (!navigation.selected.includes(i)) {
-          show.push(i);
+    const config = Selectors.config(state);
+    const max = ConfigAdapter.extractMaxRefinements(config);
+    const booleanNavs = ConfigAdapter.extractToggleNavigations(config);
+    return navigations.map((navigation) => {
+      const bool = booleanNavs.includes(navigation.field);
+      if (max) {
+        const show = navigation.selected.slice(0, max);
+        for (let i = 0; i < navigation.refinements.length && show.length < max; i++) {
+          if (!navigation.selected.includes(i)) {
+            show.push(i);
+          }
         }
-      }
 
-      return {
-        ...navigation,
+        return {
```

Comment:
Since `{ ...navigation, boolean: bool }` is the same between both cases, perhaps it's worth it to extract that into a variable and add `more` and `show` only if `max` exists, then returning the resulting navigation. This would also eliminate the `else` case and make it more flexible in case we need to add more properties in the future.